src/test/java/io/aiven/elasticsearch/storage/security/RsaKeysReaderTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.elasticsearch.storage.security;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import io.aiven.elasticsearch.storage.RsaKeyAwareTest;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class RsaKeysReaderTest extends RsaKeyAwareTest {
+
+    @Test
+    public void failsForWrongPaths() {
+
+        assertThrows(
+                IllegalArgumentException.class,
+                () -> RsaKeysReader.readRsaKeyPair(
+                        Files.newInputStream(Paths.get(".")),
+                        Files.newInputStream(Paths.get(".")))
+        );
+    }
+
+    @Test
+    void throwsIllegalArgumentExceptionForEmptyPublicKey() {
```

Comment:
It seems to me that this doesn't test for the empty key, but for a wrong path again (like in `failsForWrongPaths`). Or I miss something?
The same goes for the `throwsIllegalArgumentExceptionForEmptyPrivateKey`

Reply:
true changed behave + added addition test for unsupported algo